turbo/jsonrpc/otterscan_api.go

```diff
@@ -269,19 +270,29 @@ func (api *OtterscanAPIImpl) searchTransactionsBeforeV3(tx kv.TemporalTx, ctx co
 		// Internal search code considers blockNum [including], so adjust the value
 		fromBlockNum--
 	}
-	fromTxNum, err := rawdbv3.TxNums.Max(tx, fromBlockNum)
-	if err != nil {
-		return nil, err
+	fromTxNum := -1
+	if fromBlockNum != 0 {
+		// from == 0 == magic number which means last; reproduce bug-compatibility for == 1
+		// with e2 for now
+		_txNum, err := rawdbv3.TxNums.Max(tx, fromBlockNum)
+		if err != nil {
+			return nil, err
+		}
+		fromTxNum = int(_txNum)
 	}
-	itTo, err := tx.IndexRange(kv.TracesToIdx, addr[:], int(fromTxNum), -1, order.Desc, kv.Unlim)
+
+	// ensure result union will always have at least pageSize+1 results in order to determine
+	// there is more results or this is the last page, at the same time avoiding unbounded
+	lookupSize := int(pageSize) + 1
```

Comment:
each block has: 
- 1 system txn in the beginning of block
- 1 system txn in the end of block
We added them to attribute there Rewards, and other consensys-made state changes, etc...

In next `for` loop it does `continue` such transactions. Means web page may have less rows than requested here. But maybe it's ok.

Reply:
@wmitsuda ping

Reply:
I think I misunderstood the meaning of isFinalTxn return value, will fix it

Reply:
actually lookSize was wrong, I should've let iterators unbounded + isFinalTxn will skip matching block rewards [not real txs], it is now fixed in the next push.